tests/test_reply_function_async.js

```diff
@@ -51,60 +49,30 @@ test('reply takes a callback for status code', async t => {
   scope.done()
 })
 
-test('reply should throw on error on the callback', t => {
-  let dataCalled = false
-
-  const scope = nock('http://example.com')
+test('reply should throw on error on the callback', async t => {
+  nock('http://example.test')
     .get('/')
     .reply(500, (path, requestBody, callback) =>
       callback(new Error('Database failed'))
     )
 
-  // TODO When this request is converted to `got`, it causes the request not
-  // to match.
-  const req = http.request(
-    {
-      host: 'example.com',
-      path: '/',
-      port: 80,
-    },
-    res => {
-      t.equal(res.statusCode, 500, 'Status code is 500')
-
-      res.on('data', data => {
-        dataCalled = true
-        t.ok(data instanceof Buffer, 'data should be buffer')
-        t.ok(
-          data.toString().indexOf('Error: Database failed') === 0,
-          'response should match'
-        )
-      })
-
-      res.on('end', () => {
-        t.ok(dataCalled, 'data handler was called')
-        scope.done()
-        t.end()
-      })
-    }
-  )
-
-  req.end()
+  t.rejects(got('http://example.test'), {
```

Comment:
Does this need an `await`? I think so.

```suggestion
  await t.rejects(got('http://example.test'), {
```